### PR TITLE
SUPDATA-135,138 improve colour contrast and focus styling

### PR DESCRIPTION
**Pull request by catherine-mediasuite**
[SUPDATA-136- Content obscured at high zoom levels ](https://mediasuite.atlassian.net/browse/SUPDATA-136)
- Updated the scroll to top bottom to only be visible at the very bottom of the page and for a little bit of scrolling up.

[SUPDATA-135- Keyboard focus / hover styling ](https://mediasuite.atlassian.net/browse/SUPDATA-135)
- Include accessible variations of blue, orange, green (contrast ratio 3.2:1) and use for under headings
- Update hover colour from text to darker gray
- Change two other colours (blue on a light grey background) which failed so now grey is lighter and blue is slightly darker.

[SUPDATA-138- colour contrast](https://mediasuite.atlassian.net/browse/SUPDATA-138)
- Update button and tags to have extra coloured border on hover and focus

[SUPDATA-137- Input focus colour](https://mediasuite.atlassian.net/browse/SUPDATA-137)
- Change input to darker gray on hover/focus, keeping the box shadow too.
- This should be enough as the focus cursor in the input is also black/had enough contrast.

[SUPDATA-146 - correctly implement disclosure widget ](https://mediasuite.atlassian.net/browse/SUPDATA-146)
- Update the slide function to remove ariaexpanded
- Added new toggle option for aria-label, updated examples to show aria-expanded on controller, added aria-controls, and aria-label to accordions.

[SUPDATA-144- home icon colour ](https://mediasuite.atlassian.net/browse/SUPDATA-144)
- Updated icon to a consistent dark gray regardless of theme.

### CODE REVIEW

**Review by jooleearr — CHANGES_REQUESTED** [review]
K so it's a bit hard to tell without testing (so please just let me know if I've got this wrong) but I can't see that this has updated the hover colour of the main menu items to a colour that has a sufficient contrast for text that size. 

My understanding of these code changes is that you've updated the 3 problematic theme colours to #019EAD (bondi blue), #4CA22F (green apple) and #E46911 (christine) which will be fine for the icons but does it address the issue shown here: 

image

The bondi blue colour for instance would only pass for large text (so bigger than 18px if bold and bigger than 24px if regular weight) but I believe the heading text is only 16px.

All the other issues look good to go - just any text below 18px/bold or 24px/regular using those theme colours that will still fail I think.

**Comment by catherine-mediasuite** [comment]
I changed the hover colour of the text to be a dark grey instead of theme-colour, the colour is only used for border or background colour. Shall I change so the colours have a 4.5:1 ratio instead?

**Review by jooleearr — APPROVED** [review]

File: src/6-information/accordion/_index.scss [hunk 1]
```diff
@@ -12,7 +12,7 @@
 //           Showcase
 //         </a>
 //         <div class="accordion__actions">
-//           <button class="button-icon size-sm" data-click="dirty; class: state-standard, state-active;  trigger: #group-57, #group-57-icon; ariaLabel: 'Show more showcase', 'Show less showcase'">
+//           <button class="button-icon size-sm" data-click="dirty; class: state-standard, state-active; trigger: #group-57, #group-57-icon; ariaexpanded: 'true', 'false'; arialabel: See more {$Title}, See less {$Title}" aria-controls=group-57 aria-expanded="false">
```
Comment (jooleearr) [inline src/6-information/accordion/_index.scss:15]: Think I missed this in the other PR but do the arialabel values need to be wrapped in quotes (like the ariaexpanded one)? Or do they behave differently?

File: src/6-information/checklist/_index.scss [hunk 2]
```diff
@@ -34,7 +34,7 @@
 //       </div>
 //     </div>
 //     <div class="checklist__actions">
-//       <button class="button-icon" data-click="class: state-standard, state-active;  trigger: #group-1, #group-1-content, #group-1-icon">
+//       <button class="button-icon" data-click="class: state-standard, state-active;  trigger: #group-1, #group-1-content, #group-1-icon; ariaexpanded: 'true', 'false'; arialabel: Show guidance on question 1, Hide guidance on question 1" aria-expanded="false" aria-controls="group-1-content"">
```
Comment (jooleearr) [inline src/6-information/checklist/_index.scss:37]: As above, do the arialabel values need to be wrapped in quotes?

File: src/6-information/checklist/_index.scss [hunk 3]
```diff
@@ -71,7 +71,7 @@
 //       </div>
 //     </div>
 //     <div class="checklist__actions">
-//       <button class="button-icon" data-click="class: state-standard, state-active;  trigger: #group-2, #group-2-content, #group-2-icon">
+//       <button class="button-icon" data-click="class: state-standard, state-active;  trigger: #group-2, #group-2-content, #group-2-icon; ariaexpanded: 'true', 'false'; arialabel: Show guidance on question 2, Hide guidance on question 2" aria-expanded="false" aria-controls="group-2-content">
```
Comment (jooleearr) [inline src/6-information/checklist/_index.scss:74]: Same as above :)

File: src/7-layout/header/_index.scss [hunk 4]
```diff
@@ -6,7 +6,7 @@
 // <header class="site-header">
 //   <div class="container site-header__nav-container">
 //     <div class="site-header__inner">
-//       <button class="button-icon navigation-toggle navicon" aria-label="Open menu" data-click="trigger: #site-navigation; class: in-active, is-active; ariaLabel: Open menu, Close menu">
+//       <button class="button-icon navigation-toggle navicon" aria-label="Open menu" data-click="trigger: #site-navigation; class: in-active, is-active; arialabel: Open menu, Close menu">
```
Comment (jooleearr) [inline src/7-layout/header/_index.scss:9]: Same as above :)

File: src/9-plugins/toggler/options/index.js [hunk 5]
```diff
@@ -14,6 +14,6 @@ export default {
   html,
   trigger,
   ariaexpanded,
-  ariaLabel,
+  arialabel,
```
Comment (jooleearr) [inline src/9-plugins/toggler/options/index.js]: I know this hurt, but yay for consistency ;)

**Review by jooleearr — APPROVED** [review]
Just a question around wrapping values in quotes but looks good apart from that. Again, haven't tested yet but let me know if you'd like me to before it goes through to UAT

File: src/6-information/accordion/_index.scss (hunk 1 above)
Comment (catherine-mediasuite) [reply src/6-information/accordion/_index.scss:15]: They all come through as strings without the quote marks so I could remove them.

File: src/9-plugins/back-to-top/index.js [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    topTrigger.addClass("hidden");
+  };
+
+  const showTopTrigger = () => {
+    topTrigger.removeClass("hidden");
+  };
+
+  function onScroll(currentScrollTop) {
+    const t = $(this);
+    console.log(t.scrollTop());
```
Comment (jooleearr) [inline src/9-plugins/back-to-top/index.js:17]: Might just want to remove the console log before this goes to production ;)

File: src/9-plugins/back-to-top/index.js [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+    // only show scroll to top button at the very bottom and little bit of scrolling up.
+    if (
+      t.scrollTop() + t.height() == $(document).height() ||
+      t.scrollTop() + 150 + t.height() > $(document).height()
```
Comment (jooleearr) [inline src/9-plugins/back-to-top/index.js:21]: 150 seems like a bit of a [magic number](https://hireamir.com/blog/software-anti-patterns-magic-numbers). Not a biggie here as the comment above kind of tells the story but can be good practice to assign arbitrary numbers like this to named variables that describe what they are/do :)

**Review by jooleearr — APPROVED** [review]
Looks good! I'm not sure if they're going to split hairs over this but the top button does cover up a smidge of the legal text at 400% for me still (see screenshot). If that's not an issue then :shipit:

Screenshot 2023-07-05 at 10 07 13 am

**Comment by catherine-mediasuite** [comment]
I think the button covers minimal content on scrolling up before disappearing so this text can be read. Will leave as is for now.

**Pull request merged**